design-documents/database/declarative-schema-database-compatibility.md

```diff
@@ -0,0 +1,1093 @@
+     */
+    public function getPrecision(): int {}
+
+    /**
+     * Returns the default value of the column.
+     *
+     * @return int|null
+     */
+    public function getDefaultValue(): ?int {}
+}
+```
+
+```
+class InformationSchema\Table\SmallIntColumn extends Column
+{
+    public const TYPE = 'SMALLINT';
+
+    /**
+     * Returns whether the column contains signed or unsigned integers.
+     *
+     * Note that not all RDBMS support this, thus `false` should be returned for RDBMS that do not support this.
+     *
+     * @return bool
+     */
+    public function isUnsigned(): bool {}
+
+    /**
+     * Returns the (left) padding for the integer.
+     *
+     * Note that not all RDBMS support this, returning `null` indicates whether this is supported.
+     *
+     * @return int
+     */
+    public function getPadding(): ?int {}
+
+    /**
+     * Returns the precision (amount of digits) the column can store.
+     *
+     * @return int
+     */
+    public function getPrecision(): int {}
+
+    /**
+     * Returns the default value of the column.
+     *
+     * @return int|null
+     */
+    public function getDefaultValue(): ?int {}
+}
+```
+
+```
+class InformationSchema\Table\BigIntColumn extends Column
+{
+    public const TYPE = 'BIGINT';
+
+    /**
+     * Returns whether the column contains signed or unsigned integers.
+     *
+     * Note that not all RDBMS support this, thus `false` should be returned for RDBMS that do not support this.
+     *
+     * @return bool
+     */
+    public function isUnsigned(): bool {}
+
+    /**
+     * Returns the (left) padding for the integer.
+     *
+     * Note that not all RDBMS support this, returning `null` indicates whether this is supported.
+     *
+     * @return int
+     */
+    public function getPadding(): ?int {}
+
+    /**
+     * Returns the precision (amount of digits) the column can store.
+     *
+     * @return int
+     */
+    public function getPrecision(): int {}
+
+    /**
+     * Returns the default value of the column.
+     *
+     * @return int|null
+     */
+    public function getDefaultValue(): ?int {}
+}
+```
+
+```
+class InformationSchema\Table\DecimalColumn extends Column
+{
+    public const TYPE = 'DECIMAL';
+
+    /**
+     * Returns the precision (amount of digits) the column can store.
+     *
+     * @return int
+     */
+    public function getPrecision(): int {}
+
+    /**
+     * Returns the scale (amount of digits right of the decimal point) the column can store.
+     *
+     * In case the column has not been defined to have a numeric scale (aka dynamic scale), `null` is returned.
+     *
+     * @return int|null
+     */
+    public function getScale(): ?int {}
+
+    /**
+     * Returns the default value of the column.
+     *
+     * @return float|null
+     */
+    public function getDefaultValue(): ?float {}
+}
+```
+Note that ANSI-SQL defines the exact numerical data types `DECIMAL` and `NUMERIC` as being equivalent. MySQL / MariaDB
```

Comment:
I would try to make columns types generic and delegate responsibility of mapping them to particular database engine to management classes.

Reply:
@melnikovi Agreed

Reply:
It's ok to not include this in the design. The idea is that providers will map database specific types to generic types in application.

Reply:
I added a comment in the section "Providers" stating what we have discussed.